### ISSUE TITLE
Syntastic fails to load with nonsensical error when umask has no user execute permission

### ISSUE THREAD

**Issue by sobjornstad** [scrooloose/syntastic#1781]
I have a shell script which at some point does something like:
```sh
umask 0177
"$EDITOR" "$somefile"
```
After installing Syntastic, when I run this script, I get the following strange error:

    syntastic: error: your shell /bin/bash can't handle traditional UNIX syntax for redirections

Obviously, this error makes no sense since `bash` can in fact handle normal redirection syntax.

This looks like the code immediately displaying the error:

```vim
if s:_running_windows
    let g:_SYNTASTIC_UNAME = 'Windows'
elseif executable('uname')
    try
        let g:_SYNTASTIC_UNAME = split(syntastic#util#system('uname'), "\n")[0]
    catch /\m^Vim\%((\a\+)\)\=:E484/
        call syntastic#log#error("your shell " .  syntastic#util#var('shell') . " can't handle traditional UNIX syntax for redirections")
        finish
    catch /\m^Vim\%((\a\+)\)\=:E684/
        let g:_SYNTASTIC_UNAME = 'Unknown'
    endtry
else
    let g:_SYNTASTIC_UNAME = 'Unknown'
endif
lockvar g:_SYNTASTIC_UNAME
```

The important thing appears to be the lack of execute permission in the umask (if I set the umask to 0077, it works fine). I'd be interested to know if there's actually a good reason why execute permissions are required in the umask, but assuming that there is, a more useful error message would be much appreciated.

If it matters, I'm using terminal vim 7.4 and Syntastic 3.7.0-140, on Linux Mint 17.2.

**Comment by lcd047** [comment]
> Obviously, this error makes no sense since `bash` can in fact handle normal redirection syntax.

The point of the `uname` test is to find out whether Vim can run external programs with `system()` and expect meaningful results back.  The message is of course bullshit, :smile: as Vim has no way to know why `uname` can't be run, but it does reflect the most probable cause (made so by your fellow Linux users): running Vim under `fish`, and / or messing (usually for no reason) with Vim options relevant to `system()`.  If you can come up with a message that would convey that information, I'm open to suggestions.

> I'd be interested to know if there's actually a good reason why execute permissions are required in the umask,

It isn't that syntastic _requires_ a certain `umask` (it actually doesn't care), the problem is your `umask` prevents Vim's `system()` from working.  If you disable syntastic and run `:echo system('ls')` with your `umask` you'll get back a message like this:
```
E484: Can't open file /tmp/v9YyYju/2
```
Vim creates a directory for its temporary files, than it can't access the files inside.  It probably should set `umask` to something useful before creating the directory etc., but it doesn't.  I suppose you should report this asa  bug to `vim_dev`.  In the mean time either don't do that, or set syntastic to passive mode. _shrug_

**Comment by lcd047** [comment]
The problem has been fixed in [Vim 7.4.1889](https://github.com/vim/vim/commit/35d88f4e2ff5dcd9904f04612d5febede996137c).

**Comment by sobjornstad** [comment]
Awesome, thanks for that!
